subsys/net/l2/ethernet/dsa/dsa_tag.c

```diff
@@ -36,3 +40,24 @@ struct net_pkt *dsa_tag_xmit(struct net_if *iface, struct net_pkt *pkt)
 
 	return dsa_switch_ctx->dapi->xmit(iface, pkt);
 }
+
+void dsa_tag_setup(const struct device *dev_cpu)
+{
+	const struct dsa_port_config *cfg = dev_cpu->config;
+	struct dsa_switch_context *dsa_switch_ctx = dev_cpu->data;
+
+	switch (cfg->tag_proto) {
+#ifdef CONFIG_DSA_TAG_PROTOCOL_NETC
```

Comment:
We could enhance and make this more generic in the future by having a macro that setups the tag for a vendor and places the information to a linker section. Then the generic setup function would just traverse the dsa tag section and config things accordingly.
This way we would not need to edit this file at all if other vendor protocols are added and this would also support out of tree vendor dsa tags. Anyway, this is something that can be implemented in the future.

Reply:
Thanks @jukkar for the good suggestion.
That's a good way to manage tag protocols. I will try to convert to this way in the future.